pkgs/development/mobile/xcodeenv/compose-xcodewrapper.nix

```diff
@@ -1,19 +1,39 @@
-{ stdenv, lib }:
-{ version ? "11.1"
-, allowHigher ? false
-, xcodeBaseDir ? "/Applications/Xcode.app" }:
+{ lib,
+  stdenv,
+  writeShellScriptBin }:
+{ versions ? [ "14.3" "15.1" "15.2" "15.3" ], xcodeBaseDir ? "/Applications/Xcode.app" }:
```

Comment:
How did you select these specific versions?

Reply:
These are the last few working versions and they are considered only when versions parameter is not provided.

Reply:
It's not as if older versions of Xcode stop working when new versions are released, so at least older versions should be accepted.

Reply:
they would be, you can pass whatever you wish in version argument, this list is just a fallback for when you don't pass an argument

Reply:
The original was version 11, that was either just the version of Xcode on the system at that time or it was an attempt to keep it compatible with the 11.0 SDK that aarch64-darwin stdenv is based on.

Constraining to other versions by default just doesn't seem helpful.

Reply:
I don't mind adding Xcode 11 to the list, as this is just a fallback when no version list is provided.